docs/platforms/aws-lambda/nodejs/index.md

```diff
@@ -0,0 +1,465 @@
+# AWS Lambda: Node.js
+
+AWS Lambda is a serverless compute service that lets you run code without provisioning or managing servers.  You can measure Metrics, Traces and Logs of AWS Lambda via Serverless Console.  
+
+# Tracing
+
+If you use Serverless Console's AWS Lambda Extension for Node.js, it will automatically trace your AWS Lambda invocations, providing a timeline rich with information we curated to help you assess, optimize and troubleshoot, as well as trace services and data that triggered your AWS Lambda functions.
+
+Before the handler of your AWS Lambda function runs, our Extension loads our SDK and performs the auto-instrumentation logic.  This is what it looks like:
+
+```javascript
+import serverless from 'serverless-sdk-aws-lambda'
+
+serverless.instrument(params)
+
+// Your handler is called next...
+```
+
+Every AWS Lambda function invocation instrumented with our Extension generates a Trace.  This Trace contains the following Spans, some of which are optional depending on the modules you use within your code.  We put great effort into enriching this Trace with Tags useful for debugging across code, AWS Services and more.
+
+Here is a table of contents of Spans we currently capture, in an example hierarchical format:
+
+* **aws.lambda**
+  * **aws.lambda.initialization**
+  * **aws.lambda.invocation**
+    * **express**
+      * **node.https**
+      * **node.http**
+      * **aws.sdk**
+
+# Spans
+
+## `aws.lambda`
+
+This is the parent Span (aka the Trace) for an AWS Lambda. It measures the combined lifecyle phases of AWS Lambda Initialization and Invocation, and any logic performed within the Invocation phase.
```

Comment:
Right.  You are speaking specifically of retries.  This is great feedback, and a place where users often need help.

I'm not sure I agree it should be within the same span because it is an additional invocation and reporting everything under one span is a happenstance that may affect our product experience and the clarity of what is happening.  For example, this will diverge from everything else which will indicate additional Invocations happened (our Metrics, the Metrics of other triggering infrastructure, etc.).  Numbers in charts will no longer match and I bet users will be more confused.

However, I do believe we can do something in the UX to link Traces that belong to a retry.  The user should be able to see the retry attempt number, and easily access the Trace of each retry. This would be hugely valuable.

Therefore, we should re-initialize everything, except somehow track a retry number and the Trace IDs of those retries, if possible.  Let me try and draft this out.

Reply:
Ok, so if I understand correctly, `aws.lambda` is to track a single invocation which in case of a cold start, will be accompanied by initialization that comes before.

If that's the case, then it doesn't perfectly fit the AWS Lambda lifecycle model. e.g. there can be lambda instances that were never invoked (so no request id) yet were initialized and possibly even errored, e.g. provisioned concurrency will trigger that.
Do we plan somehow to monitor such cases?

Also, shutdown phase in success cases will happen minutes after invocation (after lambda gets stale) so it's hard to attribute it to e.g. last invocation.

Reply:
`aws.lambda` is intended to be the parent span, referring to handling a _request_ in Lambda.  There is nothing that accompanies it. There are only child spans that it contains, which are:

* `aws.lambda.initialization`
* `aws.lambda.invocation`

> If that's the case, then it doesn't perfectly fit the AWS Lambda lifecycle model. e.g. there can be lambda instances that were never invoked (so no request id) yet were initialized and possibly even errored, e.g. provisioned concurrency will trigger that.

`aws.lambda` and `aws.lambda.initialization` would still exist in this case.  That is, if we can still capture those in the above case.

>Also, shutdown phase in success cases will happen minutes after invocation (after lambda gets stale) so it's hard to attribute it to e.g. last invocation.

Shutdown has been ignored as a span because of this.